### PR TITLE
Align services and configurations more

### PR DESCRIPTION
**Pull request by MarcelMue**

### CODE REVIEW

**Review by tuommaki — APPROVED** [review]

**Review by TheoBrigitte — APPROVED** [review]
LGTM

**Review by tfussell — APPROVED** [review]

File: service/health/searcher/service.go [hunk 1]
```diff
@@ -10,37 +10,42 @@ import (
...
-		return nil, microerror.Maskf(invalidConfigError, "logger must not be empty")
+		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)
 	}
+
 	if config.Provider == "" {
-		return nil, microerror.Maskf(invalidConfigError, "provider must not be empty")
+		return nil, microerror.Maskf(invalidConfigError, "%T.Provider must not be empty", config)
 	}
 
 	newService := &Service{
```
Comment (kopiczko) [inline service/health/searcher/service.go]: IMO this should have the same name as in the method receiver:

```suggestion
	s := &Service{
```

File: service/health/searcher/service.go [hunk 2]
```diff
@@ -10,37 +10,42 @@ import (
...
-		Config: config,
+		g8sClient: config.G8sClient,
+		logger:    config.Logger,
+
+		provider: config.Provider,
 	}
 
 	return newService, nil
```
Comment (kopiczko) [inline service/health/searcher/service.go]:
```suggestion
	return s, nil
```

**Review by kopiczko — APPROVED** [review]

**Comment by tfussell** [comment]
Towards https://github.com/giantswarm/giantswarm/issues/6893

**Pull request merged**